float32 position 
---
float32 total_time
